core/src/main/java/com/datastax/oss/driver/api/core/session/SessionBuilder.java

```diff
@@ -223,6 +225,18 @@ public SelfT withRequestTracker(@Nullable RequestTracker requestTracker) {
     return self;
   }
 
+  /**
+   * Register an auth provider
+   *
+   * <p>If the auth provider is specified programmatically with this method, it overrides the
+   * configuration (that is, the {@code advanced.auth-provider} option will be ignored).
+   */
+  @NonNull
+  public SelfT withAuth(@Nullable AuthProvider authProvider) {
```

Comment:
Maybe we could also add `withPlainTextCredentials(String username, String password)` that wraps the creation and registration of a `RuntimePlainTextAuthProvider`. That way the user doesn't have to deal with an internal class.

It would come with the following caveats:
* only works with OSS Cassandra's plain text authenticator
* does not support changing the credentials at runtime
* stores the credentials as strings in memory